### PR TITLE
Enable cell editing and updating the table state

### PR DESCRIPTION
**Pull request by emirror-de**
This PR contains the updates that are required for https://github.com/Synphonyte/leptos-struct-table/pull/4. Have a look there to get more information about the changes.

### CODE REVIEW

File: src/table.rs [hunk 1]
```diff
@@ -497,6 +520,8 @@ impl ToTokens for TableComponentDeriveInput {
                     // on_row_click(event);
                 };
 
+                let row_data_provider = items.clone();
```
Comment (maccesch) [inline src/table.rs]: this doesn't seem to be used?

**Review by maccesch — CHANGES_REQUESTED** [review]

File: src/table.rs (hunk 1 above)
Comment (emirror-de) [reply src/table.rs]: It is not directly used in this function, but required to transfer the `items` to the `on_cell_change` closure in line `124`. I did not see a better way to do it.

Comment (maccesch) [reply src/table.rs]: Oh yeah, seeing it now. 🤦🏻‍♂️ Sorry
I think it's ok for now, the component isn't prepared for large datasets yet anyway.

File: src/table.rs [hunk 2]
```diff
@@ -572,6 +597,9 @@ impl ToTokens for TableComponentDeriveInput {
 
                                                 let selected_signal = Signal::derive(cx, move || is_sel(Some(item.#key_field.clone())));
 
+                                                // required to be able to get `item` into on_cell_change
+                                                let (row_state, _) = create_signal(cx, item.clone());
```
Comment (maccesch) [inline src/table.rs]: I just remembered that there is a leptos primitive made exactly for this: 
https://docs.rs/leptos/latest/leptos/fn.store_value.html

This saves the reactivity overhead.

**Review by maccesch — CHANGES_REQUESTED** [review]

File: src/table.rs (hunk 2 above)
Comment (emirror-de) [reply src/table.rs]: Awesome, I was not aware of this yet. Updated in 7ae1d1e

**Comment by maccesch** [comment]
This is already implemented in the meantime. Thanks!

**Pull request closed without merging**